File: quick-rescue/frontend/src/services/familiar.service.ts
import { api } from './api';
import { Familiar, ApiResponse } from '../types';

export const listar = async (): Promise<Familiar[]> => {
  const { data } = await api.get<ApiResponse<Familiar[]>>('/familiares');
  return data.data!;
};

export const crear = async (payload: Omit<Familiar, 'id' | 'usuario_id' | 'creado_en'>): Promise<Familiar> => {
  const { data } = await api.post<ApiResponse<Familiar>>('/familiares', payload);
  return data.data!;
};

export const actualizar = async (id: number, payload: Partial<Familiar>): Promise<Familiar> => {
  const { data } = await api.put<ApiResponse<Familiar>>(`/familiares/${id}`, payload);
  return data.data!;
};

export const eliminar = async (id: number): Promise<void> => {
  await api.delete(`/familiares/${id}`);
};

File: mobile/lib/services/familiar_service.dart
import '../models/familiar.dart';
import 'api_service.dart';

class FamiliarService {
  static Future<List<Familiar>> listar() async {
    final data = await ApiService.get('/familiares') as List<dynamic>;
    return data.map((e) => Familiar.fromJson(e as Map<String, dynamic>)).toList();
  }

  static Future<Familiar> crear(Map<String, dynamic> payload) async {
    final data = await ApiService.post('/familiares', payload) as Map<String, dynamic>;
    return Familiar.fromJson(data);
  }

  static Future<Familiar> actualizar(int id, Map<String, dynamic> payload) async {
    final data = await ApiService.put('/familiares/$id', payload) as Map<String, dynamic>;
    return Familiar.fromJson(data);
  }

  static Future<void> eliminar(int id) async {
    await ApiService.delete('/familiares/$id');
  }
}
